agent/engine/docker_task_engine.go

```diff
@@ -56,7 +62,8 @@ type DockerTaskEngine struct {
 	taskEvents      chan api.TaskStateChange
 	saver           statemanager.Saver
 
-	client DockerClient
+	client     DockerClient
+	clientLock sync.Mutex
```

Comment:
I think this would read better if it were a sync.Once instead, and would be functionally equivalent.